### PR TITLE
Discovery schema

### PR DESCRIPTION
**Pull request by Krishnan-Priyanshu**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

### CODE REVIEW

**Comment by terraform-providers-svc** [comment]
Can one of the admins verify this patch?

File: ome/resource_discovery_schema.go [hunk 1]
```diff
@@ -0,0 +1,469 @@
...
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// DiscoveryJobSchema for discovery job schema.
+func DiscoveryJobSchema() map[string]schema.Attribute {
+	return map[string]schema.Attribute{
+
+		"discovery_job_id": schema.Int64Attribute{
```
Comment (rounak-adhikary) [inline ome/resource_discovery_schema.go:16]: schemas require an "id" field. Does it make sense to rename this to "id"?

File: ome/resource_discovery_schema.go [hunk 2]
```diff
@@ -0,0 +1,469 @@
...
+
+		"email_recipient": schema.StringAttribute{
+			MarkdownDescription: `
+				- Enter the email address to which notifications are to be sent about the discovery job status.
+				- Configure the SMTP settings to allow sending notifications to an email address.`,
+			Description: `
+				- Enter the email address to which notifications are to be sent about the discovery job status.
+				- Configure the SMTP settings to allow sending notifications to an email address.`,
+			Optional: true,
+			Computed: true,
```
Comment (rounak-adhikary) [inline ome/resource_discovery_schema.go:39]: Please confirm if this is supposed to be computed as well.

File: ome/resource_discovery_schema.go [hunk 3]
```diff
@@ -0,0 +1,469 @@
...
+			MarkdownDescription: "The maximum wait time of job_wait in seconds. The job is tracked only for this duration.",
+			Description:         "The maximum wait time of job_wait in seconds. The job is tracked only for this duration.",
+			Optional:            true,
+			Computed:            true,
+			PlanModifiers: []planmodifier.Int64{
+				Int64DefaultValue(types.Int64Value(10800)),
+			},
+		},
+
+		"schedule": schema.StringAttribute{
```
Comment (rounak-adhikary) [inline ome/resource_discovery_schema.go:76]: Can't this just be a bool attribute?

File: ome/resource_discovery_schema.go (hunk 1 above)
Comment (Krishnan-Priyanshu) [reply ome/resource_discovery_schema.go]: renamed to id

File: ome/resource_discovery_schema.go [hunk 4]
```diff
@@ -0,0 +1,469 @@
...
+func DiscoveryJobSchema() map[string]schema.Attribute {
+	return map[string]schema.Attribute{
+
+		"discovery_job_id": schema.Int64Attribute{
+			MarkdownDescription: "ID of the discovery configuration group",
+			Description:         "ID of the discovery configuration group",
+			Computed:            true,
+		},
+
+		"discovery_job_name": schema.StringAttribute{
```
[the remark this answers is not in the document]
Comment (Krishnan-Priyanshu) [reply ome/resource_discovery_schema.go]: rename it to name

File: ome/resource_discovery_schema.go (hunk 3 above)
Comment (Krishnan-Priyanshu) [reply ome/resource_discovery_schema.go:75]: we are keeping schedule as string attribute with two choices runnow and runlater

File: ome/resource_discovery_schema.go (hunk 2 above)
Comment (Krishnan-Priyanshu) [reply ome/resource_discovery_schema.go]: remove the computed as true from the attribute

**Review by Krunal-Thakkar — APPROVED** [review]
LGTM

**Review by rounak-adhikary — APPROVED** [review]
lgtm

**Pull request merged**